conducto/pipeline.py

```diff
@@ -0,0 +1,195 @@
+import conducto as co
+import compose
+from pathlib import Path
+import yaml
+
+# #### What is this? ####
+#
+# The docker-compose.yml at the root of this repo is set up to run
+# released bits only.  As is, it's not helpful for trying out uncommitted
+# changes.  Maybe an enirely separate docker-compose.yml for
+# in-development bits is the right way to go. So instead of doing this...
+# https://github.com/StackStorm/st2-dockerfiles/blob/d6ee9ce0735e6079e732aa762d02d459e9863851/base/Dockerfile#L42
+# ...such a file would live in the st2 repo and use COPY dockerfile
+# directives to build the local code into the services that it ran.
+#
+# That would be the way to go if somebody was willing to maintain parity
+# between the two.
+#
+# This pipeline does something altogether stranger. It uses the structure
+# defined by the release-only docker-compose.yml, and injects local code into
+# the containerized services at runtime by mounting parts of the dev's env at
+# /opt/stackstorm/st2/lib/python3.6/site-packages/*
+#
+# Between "up" and "down" it runs a very limited sanity check just to ensure
+# that the services did indeed start and to give you a gist about how you would
+# add more tests.
+
+
+# #### Usage ####
+#
+# first:
+#     - create a conducto account (they're free)
+#     - start docker
+#     - from the conducto web UI, copy the local agent start command
+#     - paste it into a shell
+#     $ pip install conducto pyyaml
+#
+# dry-run for released changes
+#     $ python conducto/pipeline.py updown
+#
+# dry-run for dev changes
+#     $ python conducto/pipeline.py updown /path/to/dev/st2
+#
+# generate and run with released-changes only:
+#     $ python conducto/pipeline.py updown --local --run
+#
+# generate and run with dev changes only:
+#     $ python conducto/pipeline.py updown /path/to/dev/st2 --local --run
+#
+# The commands above create ./docker-compose.yml.g which is ued by the pipeline
+# below, but you can also use it outside of conducto to control service state:
+#
+#     docker-compose -f docker-compose.g.yml -p foo-bar up
```

Comment:
`-p foo-bar` can be omitted unless the pipeline with id `foo-bar` deployed the services and now you're trying to frob them.